### PR TITLE
Add support for Mezzano.

### PR DESCRIPTION
**Pull request by ebrasca**

### CODE REVIEW

**Comment by sionescu** [comment]
Hi @ebrasca, before I start reviewing please squash this into a single commit.

**Comment by ebrasca** [comment]
Hi @sionescu 

I don't know how to do it!
Is it going to leave froggey without merit?

Thank you!

**Comment by sionescu** [comment]
You can use git rebase, and squash everything into two commits then.

**Comment by ebrasca** [comment]
Is this ok?

**Comment by sionescu** [comment]
Yes, thank you.

File: src/impl-mezzano.lisp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+(deftype thread ()
+  'mezzano.supervisor:thread)
+
+;;; Thread Creation
+
+(defun %make-thread (function name)
+  (mezzano.supervisor:make-thread
+   (lambda ()
+     (with-simple-restart (abort "Terminate thread ~S" (mezzano.supervisor:current-thread))
```
[the remark this answers is not in the document]
Comment (froggey) [reply src/impl-mezzano.lisp:20]: I think it was added for debugging, it can be removed.

[the remark this answers is not in the document]
Comment (ebrasca) [reply src/impl-mezzano.lisp]: I have removed this restart and squash this into a single commit.

File: src/impl-mezzano.lisp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+(deftype thread ()
+  'mezzano.supervisor:thread)
+
+;;; Thread Creation
+
+(defun %make-thread (function name)
+  (mezzano.supervisor:make-thread
+   (lambda ()
+     (funcall function))
```
Comment (sionescu) [inline src/impl-mezzano.lisp:20]: This funcall is unnecessary.

File: src/impl-mezzano.lisp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+;;; Resource contention: condition variables
+
+(defun make-condition-variable (&key name)
+  (mezzano.supervisor:make-condition-variable name))
+
+(defun condition-wait (condition-variable lock &key timeout)
+  (mezzano.supervisor:condition-wait
+   condition-variable
+   (if (recursive-lock-p lock)
```
Comment (sionescu) [inline src/impl-mezzano.lisp:82]: Conditions should always use regular mutexes.

Comment (ebrasca) [reply src/impl-mezzano.lisp:82]: I don't understand.

Comment (sionescu) [reply src/impl-mezzano.lisp:82]: The code shouldn't check for recursive locks because condition variables should only use regular locks.

**Comment by ebrasca** [comment]
I think I solved this 2 problems.

**Comment by ebrasca** [comment]
Here documentation for what mezzano.supervisor:condition-wait does : https://github.com/froggey/Mezzano/blob/master/supervisor/sync.lisp#L297

**Comment by sionescu** [comment]
Ok, then don't return T at the end of bt:condition-wait, just remove it. Are all tests passing ?

**Comment by ebrasca** [comment]
Here result of runing tests:

BORDEAUX-THREADS/TEST> (run-all-tests)

Running test suite BORDEAUX-THREADS
 Running test SEMAPHORE-SIGNAL-N-OF-M ..
 Running test WITH-TIMEOUT-NON-INTERFERENCE .
 Running test SHOULD-IDENTIFY-THREADS-CORRECTLY ...
 Running test WITH-TIMEOUT-SIGNALS .
 Running test WITH-TIMEOUT-RETURN-VALUE .
 Running test CURRENT-THREAD-IDENTITY .
 Running test SHOULD-HAVE-THREAD-INTERACTION .
 Running test SHOULD-HAVE-CURRENT-THREAD .
 Running test CONDITION-WAIT-TIMEOUT ..
 Running test SHOULD-LOCK-WITHOUT-CONTENTION ..
 Running test SEMAPHORE-WAIT-TIMEOUT ...
 Running test SHOULD-RETRIEVE-THREAD-NAME .
 Running test JOIN-THREAD-RETURN-VALUE .
 Running test INTERRUPT-THREAD .
 Running test CONDITION-VARIABLE .
 Running test SEMAPHORE-TYPED ...
 Running test SEMAPHORE-SIGNAL .
 Running test DEFAULT-SPECIAL-BINDINGS ....
 Did 30 checks.
    Pass: 30 (100%)
    Skip: 0 ( 0%)
    Fail: 0 ( 0%)

T

**Comment by sionescu** [comment]
Looks good, thanks. Merged the code.

**Pull request closed without merging**